src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,535 @@
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly CONTENTFUL_SPACE_ID: string;
+  readonly CONTENTFUL_DELIVERY_TOKEN: string;
+  readonly CONTENTFUL_PREVIEW_TOKEN: string;
+}
+```
+:::tip
+Check out [Using environment variables](/en/guides/environment-variables/) for more information about environment variables and `.env` files in Astro.
+:::
+
+Your root directory should now include these new files:
+
+```ini title="Project Structure" ins={2-3}
+├── src/
+│   └── env.d.ts
+├── .env
+├── astro.config.mjs
+└── package.json
+```
+
+### Installing dependencies
+
+To connect with your Contentful space, install [`contentful.js`](https://github.com/contentful/contentful.js), official Contentful SDK for JavaScript, and [`rich-text-html-renderer`](https://github.com/contentful/rich-text/tree/master/packages/rich-text-html-renderer), a package to render rich text fields in HTML.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with npm 
+  npm install contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with pnpm
+  pnpm install contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with yarn
+  yarn add contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Next, create a new file called `contentful.ts` in `src/lib/` directory in your project.
+
+```ts title="src/lib/contentful.ts"
+import contentful from "contentful";
+
+export const contentfulClient = contentful.createClient({
+  space: import.meta.env.CONTENTFUL_SPACE_ID,
+  accessToken: import.meta.env.DEV
+    ? import.meta.env.CONTENTFUL_PREVIEW_TOKEN
+    : import.meta.env.CONTENTFUL_DELIVERY_TOKEN,
+  host: import.meta.env.DEV ? "preview.contentful.com" : "cdn.contentful.com",
+});
+
+```
+
+The above code snippet creates a new Contentful client, passing in credentials from the `.env` file. 
+
+:::caution
+While in development mode, your content will be fetched from the **Contentful preview API**. This means that you will be able to see unpublished content from the Contentful web app. At build time, your content will be fetched from the **Contentful delivery API**. This means that only published content will be available at build time. 
+:::
+
+Finally, your root directory should now include these new files:
+
+```ini ins={3-4}
+├── src/
+│   └── env.d.ts
+│   └── lib/
+│       └── contentful.ts
+├── .env
+├── astro.config.mjs
+└── package.json
+```
+
+### Fetching data
+
+Now that you have your Contentful client set up, you can fetch data from Contentful inside your Astro components. For example, if the content type in your Contentful space has a text field for a title and a rich text for content your component may look like this:
```

Comment:
I feel like we should explain where the "blogPost" string in getEntries comes from:
```suggestion
Now that you have your Contentful client set up, you can fetch data from Contentful inside your Astro components. For example, if you have a content type with id "blogPost" that has a text field for a title and a rich text field for content, your component might look like this:
```